fixes two bugs:
1. doesn't add \ before spaces when showing dirname in interface title.
2. when replace_path() string is to be parsed by input command,
I assume that the path is to be run in shell,
and I do special escaping of 'into \'\\\'\'

diff -r f07b83b8f199 libmenu/menu_filesel.c
--- a/libmenu/menu_filesel.c	Sun Apr 30 16:54:42 2006 +0200
+++ b/libmenu/menu_filesel.c	Sun Apr 30 16:54:43 2006 +0200
@@ -84,7 +84,7 @@ static void free_entry(list_entry_t* ent
   free(entry);
 }
 
-static char* replace_path(char* title , char* dir) {
+static char* replace_path(char* title , char* dir , int escape) {
   char *p = strstr(title,"%p");
   if(p) {
     int tl = strlen(title);
@@ -94,16 +94,28 @@ static char* replace_path(char* title , 
     char *r, *n, *d = dir;
     char term = *(p-1);
 
+    if (escape) {
     do {
-      if (*d == '\\' || *d == term)
+      if (*d == '\\')
         l++;
+      else if (*d == '\'') /* ' -> \'\\\'\' */
+        l+=7;
     } while (*d++);
+    }
     r = malloc(l + 1);
     n = r + t1l;
     memcpy(r,title,t1l);
     do {
-      if (*dir == '\\' || *dir == term)
+      if (escape) {
+      if (*dir == '\\')
         *n++ = '\\';
+      else if (*dir == '\'') { /* ' -> \'\\\'\' */
+        *n++ = '\\'; *n++ = '\'';
+        *n++ = '\\'; *n++ = '\\';
+        *n++ = '\\'; *n++ = '\'';
+        *n++ = '\\';
+      }
+      }
     } while ((*n++ = *dir++));
     if(tl - t1l - 2 > 0)
       strcpy(n-1,p+2);
@@ -201,7 +201,7 @@ static int open_dir(menu_t* menu,char* a
     free(mpriv->p.title);
   p = strstr(mpriv->title,"%p");
 
-  mpriv->p.title = replace_path(mpriv->title,mpriv->dir);
+  mpriv->p.title = replace_path(mpriv->title,mpriv->dir,0);
 
   if ((dirp = opendir (mpriv->dir)) == NULL){
     mp_msg(MSGT_GLOBAL,MSGL_ERR,MSGTR_LIBMENU_OpendirError, strerror(errno));
@@ -307,7 +307,7 @@ static void read_cmd(menu_t* menu,int cm
 	char filename[fname_len];
 	char* str;
 	sprintf(filename,"%s%s",mpriv->dir,mpriv->p.current->p.txt);
-	str = replace_path(mpriv->dir_action,filename);
+	str = replace_path(mpriv->dir_action,filename,1);
 	c = mp_input_parse_cmd(str);
 	if(str != mpriv->dir_action)
 	  free(str);
@@ -341,7 +341,7 @@ static void read_cmd(menu_t* menu,int cm
       char filename[fname_len];
       char *str;
       sprintf(filename,"%s%s",mpriv->dir,mpriv->p.current->p.txt);
-      str = replace_path(mpriv->file_action,filename);
+      str = replace_path(mpriv->file_action,filename,1);
       c = mp_input_parse_cmd(str);
       if(str != mpriv->file_action)
 	free(str);
@@ -357,7 +357,7 @@ static void read_cmd(menu_t* menu,int cm
     char filename[fname_len];
     char *str;
     sprintf(filename,"%s%s",mpriv->dir,mpriv->p.current->p.txt);
-    str = replace_path(action, filename);
+    str = replace_path(action, filename, 1);
     mp_input_queue_cmd(mp_input_parse_cmd(str));
     if(str != action)
       free(str);
